framework_tvb/tvb/interfaces/web/controllers/simulator_controller.py

```diff
@@ -234,13 +235,37 @@ def __init__(self):
         self.burst_service = BurstService()
         self.simulator_service = SimulatorService()
         self.files_helper = FilesHelper()
-        self.cached_simulator_algorithm = self.flow_service.get_algorithm_by_module_and_class(
+        self.cached_simulator_algorithm = self.algorithm_service.get_algorithm_by_module_and_class(
             IntrospectionRegistry.SIMULATOR_MODULE, IntrospectionRegistry.SIMULATOR_CLASS)
 
     def _update_last_loaded_fragment_url(self, current_url):
         self.last_loaded_form_url = current_url
         common.add2session(common.KEY_LAST_LOADED_FORM_URL, self.last_loaded_form_url)
 
+    @cherrypy.expose
+    @handle_error(redirect=False)
+    @check_user
+    def cancel_or_remove_burst(self, burst_id):
+        """
+            Cancel or Remove the burst entity given by burst_id.
+            :returns 'reset-new': When currently selected burst was removed. JS will need to reset selection to a new entry
+            :returns 'canceled': When current burst was still running and was just stopped.
+            :returns 'done': When no action is required on the client.
+        """
+        burst_id = int(burst_id)
+        session_burst = common.get_from_session(common.KEY_BURST_CONFIG)
+
+        if session_burst.status == session_burst.BURST_RUNNING:
+            OperationService().stop_operation(session_burst.fk_simulation)
+            # Burst was stopped since it was running
+            return 'canceled'
```

Comment:
Is it a good idea to keep these "statuses" in an enum?

Reply:
In principle it is a very good idea to use Enums. 
In exactly this case, the only other place where these strings are used is in bursts.js function cancelOrRemoveBurst, thus it would require a manual synchronization between the two parts (py vs JS) anyway. 
I think the entire controllers + JS part would benefit greatly from a rewrite with a heavier (than plain) JS framework, thus I would leave such a refactor for those times. But good that you point this.

Reply:
In the end I dropped the return string entirely, as we don't really need it. The client side can be done smarter.